packages/fx-core/src/plugins/resource/localdebug/certificate.ts

```diff
@@ -89,11 +95,16 @@ export class LocalCertificateManager {
     return localCert;
   }
 
-  private async generateCertificate(certFile: string, keyFile: string): Promise<void> {
+  private async generateCertificate(certFile: string, keyFile: string): Promise<string> {
     // prepare attributes and extensions
     const now = new Date();
     const expiry = new Date();
-    expiry.setFullYear(expiry.getFullYear() + 1);
+    if (os.type() === "Windows_NT") {
+      expiry.setDate(expiry.getDate() + 7);
+    } else {
+      expiry.setFullYear(expiry.getFullYear() + 1);
```

Comment:
Just curious about why the expiration is still one year for macos.

Reply:
Still need user's credential on macos, so one week is a bit frequent...
I'll confirm the UX change with PM before merging this PR, so pending change may come in~

Reply:
Checked and update the UX:
- Linux & Mac: no change, keep the current UX
- Windows: do not show the confirm message, but popup a warning notification so that user can still view detailed documentation if wanted